pandas/tests/groupby/test_groupby.py

```diff
@@ -1898,6 +1871,82 @@ def test_intercept_builtin_sum(self):
         assert_series_equal(result, expected)
         assert_series_equal(result2, expected)
 
+    def test_describe(self):
+        df = DataFrame(
+            [[1, 2, 'foo'],
+             [1, np.nan, 'bar'],
+             [3, np.nan, 'baz']],
+            columns=['A', 'B', 'C'])
+        g = df.groupby('A')
+        gni = df.groupby('A', as_index=False)
+
+        # describe
+        expected_index = pd.Index([1, 3], name='A')
+        expected_col = pd.MultiIndex(levels=[['B'],
+                                             ['count', 'mean', 'std', 'min',
+                                              '25%', '50%', '75%', 'max']],
+                                     labels=[[0] * 8, list(range(8))])
+        expected = pd.DataFrame([[1.0, 2.0, np.nan, 2.0, 2.0, 2.0, 2.0, 2.0],
+                                 [0.0, np.nan, np.nan, np.nan, np.nan, np.nan,
+                                  np.nan, np.nan]],
+                                index=expected_index,
+                                columns=expected_col)
+        result = g.describe()
+        assert_frame_equal(result, expected)
+
+        expected = pd.concat([df[df.A == 1].describe().unstack().to_frame().T,
+                              df[df.A == 3].describe().unstack().to_frame().T])
+        expected.index = pd.Index([0, 1])
+        result = gni.describe()
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("interpolation", [
+        "linear", "lower", "higher", "nearest", "midpoint"])
+    @pytest.mark.parametrize("bar_vals,foo_vals", [
+        # Ints
+        ([1, 2, 3, 4, 5], [5, 4, 3, 2, 1]),
+        ([1, 2, 3, 4], [4, 3, 2, 1]),
+        ([1, 2, 3, 4, 5], [4, 3, 2, 1]),
+        # Floats
+        ([1., 2., 3., 4., 5.], [5., 4., 3., 2., 1.]),
+        # Missing data
+        ([1., np.nan, 3., np.nan, 5.], [5., np.nan, 3., np.nan, 1.]),
+        ([np.nan, 4., np.nan, 2., np.nan], [np.nan, 4., np.nan, 2., np.nan]),
+        # Timestamps
+        ([x for x in pd.date_range('1/1/18', freq='D', periods=5)],
+         [x for x in pd.date_range('1/1/18', freq='D', periods=5)[::-1]])
+    ])
+    @pytest.mark.parametrize('q', [0, .25, .5, .75, 1])
+    def test_quantile(self, interpolation, bar_vals, foo_vals, q):
+        # Fringe test case was not working as expected?
+        if (interpolation == 'nearest' and q == 0.5 and foo_vals == [
+                4, 3, 2, 1]):
+            pytest.skip("Unclear numpy expectation for nearest result with "
```

Comment:
This is extremely nuanced but I was having trouble figuring out how NumPy would consistently handle the `nearest` tiebreaker when the provided quantile falls directly in between two values.

For instance:

```
In [2]: np.percentile([1,2,3,4], 50, interpolation='nearest')
Out[2]: 3

In [3]: np.percentile([2,4], 50, interpolation='nearest')
Out[3]: 2
```

It wasn't apparent why the first example above was rounding up and the second was rounding down. I glanced at the NumPy implementation which appears to use `np.around`. In that functions' documentation it says

```
For values exactly halfway between rounded decimal values, NumPy rounds to the nearest even value
```

But again that doesn't explain the behavior above. I'm wondering if the behavior here would even be well defined across different platforms?